src/git/inflate.ml

```diff
@@ -33,6 +33,77 @@ module None = struct
 
 end
 
+module M = struct
+  open Decompress
+
+  let size_buffer   = 0x8000
+  let input_buffer  = Bytes.create size_buffer
+  let output_buffer = Bytes.create size_buffer
+  let window = Window.create ~proof:B.proof_bytes
+
+  exception Deflate_error of Deflate.error
+
+  let deflate ?(level = 4) buff =
+    let pos = ref 0 in
+    let res = Buffer.create (Cstruct.len buff) in
+
+    Deflate.bytes
+      input_buffer output_buffer
+      (fun input_buffer -> function
+      | Some max ->
+        let n = min max (min size_buffer (Cstruct.len buff - !pos)) in
+        Cstruct.blit_to_bytes buff !pos input_buffer 0 n;
+        pos := !pos + n;
+        n
+      | None ->
+        let n = min size_buffer (Cstruct.len buff - !pos) in
+        Cstruct.blit_to_bytes buff !pos input_buffer 0 n;
+        pos := !pos + n;
+        n)
+      (fun output_buffer len ->
+      Buffer.add_subbytes res output_buffer 0 len;
+      size_buffer)
+      (Deflate.default ~proof:B.proof_bytes level)
+    |> function
+    | Ok _ -> Cstruct.of_string @@ Buffer.contents res
```

Comment:
We can use a `Bigstring` instead a `Bytes` in `decompress` and use `Cstruct.of_bigarray` then in the best case. We can imagine, we allocate a `Cstruct.t` with `Cstruct.create (Cstruct.len buff)` as the result in the beginning (so, at most, the size of the input), but as I said we can do that in the _best case_ only. Sometimes, the result of `Deflate` can be upper than the input (when the input is too small and when we store the huffman tree and the header). Obviously, it's not a common case but I already seen that.

May be, a good way is to reimplement the `Buffer` module but with `Cstruct.t`. Another implementation is to start with a `Cstruct.t list`, reverse the list at the end and concat all `Cstruct.t`.